type-check.ts

```diff
@@ -150,16 +394,163 @@ export function tcDef(env : GlobalTypeEnv, fun : FunDef<null>) : FunDef<Type> {
   return {...fun, a: NONE, body: tBody};
 }
 
+// Generic classes are type-checked by instantiating their type-parameters to every valid
```

Comment:
Doesn't this effectively re-do all the work that you do in monomorphizing? If you have all the instantiated copies of the classes, why not just type-check those?